### ISSUE TITLE
Empty mangled name in mach-o header

### ISSUE THREAD

**Issue by vtky** [KJCracks/Clutch#130]
Hey Guys,

I've just dumped an app with Clutch 2.0.1 and am trying to run it. However when I run it, it crashes and syslog has the following error 

```
Empty mangled name in mach-o header: /private/var/mobile/Containers/Bundle/Application/77254E84-57B9-43C2-9864-E9635C83CF4D/DP4MOBILE-iphone_App_Store.app/Digipass
Apr 12 03:31:36: --- last message repeated 2893 times ---
```

Any idea what is going on?

**Comment by Tatsh** [comment]
Do other apps work?

**Comment by vtky** [comment]
@Tatsh Nope. Haven't come across this with other apps. I'm thinking because this app has some protections around it that is causing this issue?

**Comment by Tatsh** [comment]
Most likely. They must have some sort of way to have the app generate invalid symbol names like empty string upon decryption when it's not done in a 'normal' way, as in Clutch.

Also, they could be checking if the header is non-encrypted in the original binary. Since you have the decrypted binary (but not in full), likely you would have to now do some extra patching to get it to work.

**Comment by Tatsh** [comment]
This app https://itunes.apple.com/us/app/digipass-for-mobile-enterprise-security-iphone/id541882261?mt=8

**Comment by aaronash** [comment]
@vtky Did you find a way around this?
